pytket/pytket/wasm/wasm.py

```diff
@@ -236,8 +256,53 @@ def check_function(
         :type number_of_returns: int
         :return: true if the signature and the name of the function is correct"""
 
-        return not self._check_file or (
+        return not self.checked or (
```

Comment:
This behavior does seem a bit weird to me as well, but I'm not sure about the intention behind checking versus not checking, presumably if you are calling `check_function` you do actually want to check?

Reply:
I think this is the structure we need to be compatible with the way `add_wasm` is using the check function. When the file handler is constructed without enabling the checks, `check_function` should always return true.

I see that this is not really intuitive, previously the `self._check_file` stored the parameter given by the user, and now the `self.check` stores if the check has been run, so even if the user does not want to check the file in some situations it does get checked.

Reply:
That sounds like a good idea to me, but I am not sure how this solved the problem that the value of `filehandler.checked` might be different then the user expected it to be?